### ISSUE TITLE
FxCop rule 'CA1062:Validate arguments of public methods' should not report issues for protected methods of sealed classes

### ISSUE THREAD

**Issue by RalfKoban** [dotnet/roslyn-analyzers#2269]
#### Analyzer package
Microsoft.CodeQuality.Analyzers

#### Package Version
v2.9.1

#### Diagnostic ID
CA1062

#### Repro steps

1. Use following code
```
    public abstract class A
    {
        public bool CheckMe() => IsType(GetType());

        protected abstract bool IsType(Type type);
    }

    public sealed class B : A
    {
        protected override bool IsType(Type type) => type.Namespace == nameof(System);
    }
```
#### Expected behavior
CA1062 is not reported for `B.IsType` because it cannot be accessed from outside (except via reflection).

#### Actual behavior
CA1062 is reported for `B.IsType`

**Comment by mavasani** [comment]
https://dotnet.myget.org/feed/roslyn-analyzers/package/nuget/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.4-beta1.19317.5+45537f7f should have the fix
